### PR TITLE
Try to extract error message from non-2xx responses

### PR DESCRIPTION
**Pull request by bjorncs**

### CODE REVIEW

File: vespa-http-client/src/test/java/com/yahoo/vespa/http/client/core/communication/ApacheGatewayConnectionTest.java [hunk 1]
```diff
@@ -292,6 +303,35 @@ public void dynamic_headers_are_added_to_the_response() throws IOException, Serv
...
+                    new FeedParams.Builder().build(),
+                    "",
+                    new ConnectionParams.Builder().build(),
+                    mockFactory,
+                    "clientId");
+        apacheGatewayConnection.connect();
+        apacheGatewayConnection.handshake();
+
+        apacheGatewayConnection.writeOperations(Collections.singletonList(createDoc("42", "content", true)));
+        assertTrue(verifyContentSentLatch.await(10, TimeUnit.SECONDS));
```
Comment (vekterli) [inline vespa-http-client/src/test/java/com/yahoo/vespa/http/client/core/communication/ApacheGatewayConnectionTest.java]: I'm not sure why the tests in this class even use a `CountDownLatch`, as (if I understand the mock setup correctly) the lambda given to `mockHttpClientFactory` executes in the context of the test thread (and this isn't actually thread safe anyway). Maybe things have changed over time, or it's a fascinating way to fail the test if the lambda isn't executed (seems the most likely) ¯\\\_(ツ)_/¯ Would probably be better to use an explicit mock expectation, but this _does_ work (though confusing) so it's optional.

**Review by vekterli — APPROVED** [review]
:+1: LGTM, only have a comment re: interesting use of countdown latch

**Comment by vekterli** [comment]
Also, apologies for the belated review. This PR flew under my radar for some reason. `TooManyEmailsException`

**Review by vekterli — APPROVED** [review]
👌 merge at will

**Comment by bjorncs** [comment]
Java build completed - merging.

**Pull request merged**